llvm/utils/TableGen/RISCVTargetDefEmitter.cpp

```diff
@@ -210,10 +211,58 @@ static void emitRISCVProcs(RecordKeeper &RK, raw_ostream &OS) {
   OS << "\n#undef TUNE_PROC\n";
 }
 
+static inline uint64_t getValueFromBitsInit(const BitsInit *B,
+                                            const Record &R) {
+  assert(B->getNumBits() <= sizeof(uint64_t) * 8 && "BitInits' too long!");
+
+  uint64_t Value = 0;
+  for (unsigned i = 0, e = B->getNumBits(); i != e; ++i) {
+    const auto *Bit = dyn_cast<BitInit>(B->getBit(i));
```

Comment:
Use `cast` here. If the dyn_cast were to fail, the calculated mask would be incorrect so we should just assume it won't fail.

Reply:
`cast<BitInit>(B->getBit(i));` now